frontend/packages/pipelines-plugin/src/components/pipelines/resource-types/triggers.ts

```diff
@@ -28,29 +28,79 @@ export type TriggerTemplateKind = K8sResourceCommon & {
 
 export type EventListenerKindBindingReference = {
   // TriggerBinding / ClusterTriggerBinding reference
-  kind: string;
+  // Kind can only be provided if Ref is also provided. Defaults to TriggerBinding
+  kind?: string;
   // Ref is used since Tekton Triggers 0.5 (part of OpenShift Pipeline Operator 1.1)
-  ref: string;
+  // Mutually exclusive with Name
```

Comment:
This is a key opportunity for us to use TypeScript unions. 

Craft two types, one for old version and one for next version and then do a union of the two types:

```
/** @deprecated use EventListenerKindBindingReference instead */
type EventListenerKindBindingReferenceLegacy = {
  // Support for older operators (<0.5 Tekton Triggers)
  // https://github.com/tektoncd/triggers/pull/603/files
  // https://github.com/tektoncd/triggers/releases/tag/v0.5.0 and
  // https://github.com/tektoncd/triggers/releases/tag/v0.6.0
  name: string;
  value: string;
};

export type EventListenerKindBindingReference =
  | {
      // Kind/Ref is used since Tekton Triggers 0.5 (part of OpenShift Pipeline Operator 1.1)
      // Defaults to TriggerBinding
      kind?: string;
      ref: string;
    }
  | EventListenerKindBindingReferenceLegacy;
```

Honestly we probably won't keep this for much longer. Perhaps part of the new parity support in 4.8 we remove the legacy type.